### DISCUSSION TITLE
Docker-Compose Fail

### DISCUSSION THREAD

**Discussion by ghilasson** [firefly-iii/firefly-iii#5247] [Firefly III] [answered]
Have to resort to posting this because my brain is fried.
I've tried so many methods of installing firefly-iii on Ubuntu and failed at all (shows how bad I am at this)

Thought I was getting somewhere with docker-compose and then ran into this when I navigate to localhost:

`Could not poll the database: SQLSTATE[HY000] [1045] Access denied for user 'firefly'@'172.19.0.3' (using password: YES) (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1) `

Please could someone let me know what it is I'm doing wrong?

**Comment by JC5** [comment, marked as answer]
Is the MySQL password the same in the docker-compose file as it is in the .env file?

If yes, delete all images and all volumes and retry.

**Reply by ghilasson** [reply]
Thanks for getting back!

It's not the same no. I have a random password set in the docker-compose file (MYSQL_PASSWORD=XXXXXXX) and everything in the .env is untouched and left as default.

**Reply by JC5** [reply]
They should be the same. Otherwise Firefly III is trying to login with the original unsafe password while MySQL is configured with a strong and secure one.

**Reply by ghilasson** [reply]
Makes sense which is more than I do.

To be absolutely certain when we're talking about the password in the .env file it's the following?
DB_PASSWORD=

**Reply by ghilasson** [reply]
(Thank you very much for the help - I believe I have it running)

**Reply by JC5** [reply]
That was the one yes!
